### DISCUSSION TITLE
Root tag detection is too strict

### DISCUSSION THREAD

**Discussion by dennisprudlo** [livewire/livewire#7381] [7. Bugs (archived)]
### Livewire version

3.2.1

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create a new Laravel app with Livewire installed.
2. Create a blade component (like for a panel) with the following content
```html
@if (false)
    <div></div>
@else
    <div></div>
@endif
```
3. Create a full page component (and hook up routing) with the following content:
```html
<x-your-component>
    Foo
</x-your-component>
```

Livewire won't render the component and throws an exception with the message "Livewire encountered a missing root tag when trying to render a component. When rendering a Blade view, make sure it contains a root HTML tag." As a matter of fact in my case the blade component was a bit different:
```html
@if (condition-1)
    <div {{ $attributes }}>
@elseif (condition-2)
    </div>
@else
    <div {{ $attributes }}>{{ $slot }}</div>
@endif
```
For the Livewire component (where this blade component has been used as a root tag) both conditions were false, so there wouldn't be an issue if Livewire actually rendered the whole thing. But this here is what is causing the exception:
https://github.com/livewire/livewire/blob/367cdb0988b3efae3dfc65222a7e62cd57a0e5ed/src/Drawer/Utils.php#L13-L20

The problem is, that the `$html` variable for the contents of the blade component (in step 2) looks like this:
```html
<!--[if BLOCK]><![endif]-->    <div></div>
 <!--[if ENDBLOCK]><![endif]-->
```

I'm not sure if removing the block markers before checking has a larger impact on other parts of Livewire.

This results in other obscure bugs, but this seems to be the root of the problem. In my case I was using a panel component as a root tag with exactly the same content as my example above (if-elseif-else snippet) like this:
```html
<x-panel wire:init="someMethod">
    <x-another-component>...</x-another-component>
    <div>...</div>
</x-panel>
```

And the error I got was that Livewire executed a lazy loaded update request due to the `wire:init` attribute but in that batch update request was also an entry for the full page parent component with the `someMethod` call. This full page component does not have that method and livewire complained that the method does not exist (obviously). Only as I cleared out the slot contents of the `<x-panel>` component in the livewire component (so that an empty `<x-panel></x-panel>` remains) I got the "Missing root tag" exception.

Wrapping the `<x-panel>` completely in a `<div>` resolves the issue since livewires regex is satisfied but I wouldn't want to check all my livewire components for a root blade component and then check if that blade component has no conditions for their root tags and then figure out some way to wrap it in an unconditional html element without messing up the whole layout due to the css layout.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [x] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by dennisprudlo** [comment]
Oh and I also tried setting `inject_morph_markers' => false`. Dumping the `$html` again there are still morph markers in the HTML but it doesn't seem to be a problem anymore for the regex. However since the update request still contains an update for the full page component with the method call on a method that does not exist there I still get the "Method does not exist" exception. This must still be the same thing (Livewire not correctly rendering the `<x-panel>`) since wrapping it in a `<div>` again resolves the issue (so no matter what value `inject_morph_markers` has in the config).

Since Livewire cannot establish a component from this but somehow still renders it the `wire:init` probably looks for the closest livewire component (`wire:id`) it can find and attempts executed the method call on this component.

**Comment by dbpolito** [comment]
I'm also having this issue randomly, but in my case, no if was provided at all and i got a render method on my component:

```php
    public function render()
    {
        return <<<'BLADE'
            <div>
                <x-filament-actions::modals />
            </div>
        BLADE;
    }
```

In theory it should always have a div here, but this is throwing.
